client/container.go

```diff
@@ -346,7 +346,7 @@ func (c *container) Update(ctx context.Context, opts ...UpdateContainerOpts) err
 	return nil
 }
 
-func (c *container) Checkpoint(ctx context.Context, ref string, opts ...CheckpointOpts) (Image, error) {
+func (c *container) Checkpoint(ctx context.Context, ref, imagePath string, opts ...CheckpointOpts) (Image, error) {
```

Comment:
I don't think we need to introduce `imagePath` parameter here.
Same can already be achieved via options.

```golang
c.Checkpoint(..., WithCheckpointImagePath(imagePath))
```

Reply:
> I don't think we need to introduce `imagePath` parameter here. Same can already be achieved via options.
> 
> ```go
> c.Checkpoint(..., WithCheckpointImagePath(imagePath))
> ```

Yes... But `WithCheckpointImagePath` is for task not for container, I have simplified the code, PLZ review it again.